src/core/render.js

```diff
@@ -0,0 +1,41 @@
+//Include helper functions and Popper
+import Popper from 'popper.js';
+const createBoundingBox = require('./createBoundingBox.js');
+const createReferenceObject = require('./createReferenceObject.js');
+
+//Create a new popper object associated with a cytoscape element (Nodes or Edges)
+module.exports.createPopperObject = function (cyElement, userOptions) {
```

Comment:
It may make sense to pull out all the if-node, if-edge, if-cy bits and just have the common stuff here.  The cy stuff would go in `coreRef()` and ele stuff goes in `collectionRef()`.  You can use `ele.isNode()` in `collectionRef()` if you need